### ISSUE TITLE
[shelly] RGBW2 : Exception with when setting brightness to 0 [solved]

### ISSUE THREAD

**Issue by hpd96** [openhab/openhab-addons#6962]
shelly binding is sending "OFF" to shellyrgbw2 when brighness is set to "0"
URL:
http://192.168.42.254/relay/0?turn=off
answer of shelly is "not found" as API not supporting this

bundle:list | grep Shelly  -> 2.5.0  | Shelly Binding

## Expected Behavior
no exception
set state to OFF 

## Current Behavior
when sending "0%" to brightness channel there will be exception [NOK]. brightness will be set to 0% [OK]

```
2020-02-04 22:49:37.735 [INFO ] [e.model.script.terraSSE dimmer links] - 20
2020-02-04 22:49:38.301 [INFO ] [e.model.script.terraSSE dimmer links] - 10
2020-02-04 22:49:52.755 [INFO ] [e.model.script.terraSSE dimmer links] - 0
2020-02-04 22:49:52.806 [WARN ] [y.internal.handler.ShellyBaseHandler] - shellyrgbw2-66260d ERROR: Unable to process command for channel shelly:shellyrgbw2-white:66260d:channel1#brightness: Shelly API call failed: Unexpected http response: Not Found, url=http://192.168.42.254/relay/0?turn=off (class java.io.IOException)
Stack Trace: [org.openhab.binding.shelly.internal.api.ShellyHttpApi.request(ShellyHttpApi.java:494), org.openhab.binding.shelly.internal.api.ShellyHttpApi.setRelayTurn(ShellyHttpApi.java:134), org.openhab.binding.shelly.internal.handler.ShellyLightHandler.handleDeviceCommand(ShellyLightHandler.java:155), org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.handleCommand(ShellyBaseHandler.java:287), sun.reflect.GeneratedMethodAccessor780.invoke(Unknown Source), sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43), java.lang.reflect.Method.invoke(Method.java:498), org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152), org.eclipse.smarthome.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59), com.sun.proxy.$Proxy4565.handleCommand(Unknown Source), org.eclipse.smarthome.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:74), org.eclipse.smarthome.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:48), sun.reflect.GeneratedMethodAccessor779.invoke(Unknown Source), sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43), java.lang.reflect.Method.invoke(Method.java:498), org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152), org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:52), java.util.concurrent.FutureTask.run(FutureTask.java:266), java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149), java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624), java.lang.Thread.run(Thread.java:748)]
2020-02-04 22:49:52.845 [WARN ] [y.internal.handler.ShellyBaseHandler] - shellyrgbw2-66260d ERROR: Unable to process command for channel shelly:shellyrgbw2-white:66260d:channel2#brightness: Shelly API call failed: Unexpected http response: Not Found, url=http://192.168.42.254/relay/1?turn=off (class java.io.IOException)
Stack Trace: [org.openhab.binding.shelly.internal.api.ShellyHttpApi.request(ShellyHttpApi.java:494), org.openhab.binding.shelly.internal.api.ShellyHttpApi.setRelayTurn(ShellyHttpApi.java:134), org.openhab.binding.shelly.internal.handler.ShellyLightHandler.handleDeviceCommand(ShellyLightHandler.java:155), org.openhab.binding.shelly.internal.handler.ShellyBaseHandler.handleCommand(ShellyBaseHandler.java:287), sun.reflect.GeneratedMethodAccessor780.invoke(Unknown Source), sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43), java.lang.reflect.Method.invoke(Method.java:498), org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152), org.eclipse.smarthome.core.internal.common.InvocationHandlerSync.invoke(InvocationHandlerSync.java:59), com.sun.proxy.$Proxy4569.handleCommand(Unknown Source), org.eclipse.smarthome.core.thing.internal.profiles.ProfileCallbackImpl.handleCommand(ProfileCallbackImpl.java:74), org.eclipse.smarthome.core.thing.internal.profiles.SystemDefaultProfile.onCommandFromItem(SystemDefaultProfile.java:48), sun.reflect.GeneratedMethodAccessor779.invoke(Unknown Source), sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43), java.lang.reflect.Method.invoke(Method.java:498), org.eclipse.smarthome.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:152), org.eclipse.smarthome.core.internal.common.Invocation.call(Invocation.java:52), java.util.concurrent.FutureTask.run(FutureTask.java:266), java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149), java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624), java.lang.Thread.run(Thread.java:748)]
```

## Steps to Reproduce (for Bugs)

shelly RGBW2 is set to WHITE mode (in my case)

items:
Dimmer Lampe_Dimmer_Terrasse_westen_helligkeit  -> shelly:shellyrgbw2-white:66260d:channel1#brightness:
Dimmer Lampe_Dimmer_Terrasse_osten_helligkeit  -> shelly:shellyrgbw2-white:66260d:channel2#brightness:

rules:
1. sendCommand( Lampe_Dimmer_Terrasse_westen_helligkeit, 10 )  [OK]
2. sendCommand( Lampe_Dimmer_Terrasse_westen_helligkeit, 0 )  [NOK] 
3. sendCommand( Lampe_Dimmer_Terrasse_osten_helligkeit, 0 )  [NOK] 

## Your Environment

* OH 2.5.0 final
* ubuntu 16 LTS  intel atom
* oracle java 1.8

**Comment by markus7017** [comment]
confirmed

try this build: https://github.com/markus7017/myfiles/blob/master/shelly/org.openhab.binding.shelly-2.5.2-SNAPSHOT.jar?raw=true

**Comment by hpd96** [comment]
Hello Markus,

276 x Active   x  80 x 2.5.2.202002081026      x openHAB Add-ons :: Bundles :: Shelly Binding

looking good: no exception

however noting now a **flashing while going to OFF:**

auto-off mode (default) is ON
after going to brightness 0 there is sequence
OFF - ON (0.4 sec) - OFF 
on shelly dimmer lights visible for 400ms again

in the openhab logs is nothing to see about flashing

now i set thing shellyrgbw2 Helligkeit-Auto-EIN to OFF and added 
if (bright == 0) {	Lampe_Dimmer_Terrasse_westen.sendCommand(OFF) } 
--> flashing disappeared

--

2020-02-08 16:36:39.908 [vent.ChannelTriggeredEvent] - deconz:switch:homeserver:d0cf5efffecb75d6011000:buttonevent triggered 3002

==> /opt/openhab/userdata/logs/openhab.log <==

2020-02-08 16:36:39.914 [INFO ] [e.model.script.terraSSE dimmer links] - 0

==> /opt/openhab/userdata/logs/events.log <==

2020-02-08 16:36:39.927 [ome.event.ItemCommandEvent] - Item 'Lampe_Dimmer_Terrasse_westen_helligkeit' received command 0

2020-02-08 16:36:39.938 [ome.event.ItemCommandEvent] - Item 'Lampe_Dimmer_Terrasse_osten_helligkeit' received command 0

2020-02-08 16:36:39.948 [ome.event.ItemCommandEvent] - Item 'Lampe_Dimmer_Terrasse_westen' received command OFF

2020-02-08 16:36:39.963 [ome.event.ItemCommandEvent] - Item 'Lampe_Dimmer_Terrasse_osten' received command OFF

--

**Comment by markus7017** [comment]
will be included in PR #6985

**Comment by hpd96** [comment]
276 x Active x 80 x 2.5.2.202002081026 x openHAB Add-ons :: Bundles :: Shelly Binding

looking good: no exception

**Comment by markus7017** [comment]
included in version 2.5.4
